### PR TITLE
Enhancements to roles and lists

### PR DESCRIPTION
**Pull request by Fasse**
- New preference to enable/disable list module #505
- Rights management and date meeting configuration should be disabled for event roles #499
- Category "confirmation of participation" is not organization depending #498
- Remove rol_visible property and use cat_hidden instead #497 
- Only users with right rol_assign_roles should view invalid roles #495

### CODE REVIEW

File: adm_program/system/classes/user.php [hunk 1]
```diff
@@ -1619,6 +1619,29 @@ public function updateLoginData()
...
+     */
+    public function viewFormerRolesMembers()
+    {
+        global $gPreferences;
+
+        $returnCode = false;
+
+        if($gPreferences['lists_show_former_members'] === '1' && $this->checkRolesRight('rol_assign_roles'))
+        {
+            $returnCode = true;
```
Comment (ximex) [inline adm_program/system/classes/user.php]: directly `return true;`

File: adm_program/system/classes/user.php [hunk 2]
```diff
@@ -1619,6 +1619,29 @@ public function updateLoginData()
...
+        }
+        elseif($gPreferences['lists_show_former_members'] === '2' && $this->checkRolesRight('rol_edit_user'))
+        {
+            $returnCode = true;
```
Comment (ximex) [inline adm_program/system/classes/user.php]: directly `return true;`

File: adm_program/system/classes/user.php [hunk 3]
```diff
@@ -1619,6 +1619,29 @@ public function updateLoginData()
...
+        }
+
+        return $returnCode;
```
Comment (ximex) [inline adm_program/system/classes/user.php]: `return false;`

File: adm_program/system/classes/tableroles.php [hunk 4]
```diff
@@ -326,7 +326,7 @@ public function getDefaultList()
             return $defaultListId;
         }
 
-        if($this->getValue('cat_name_intern') == 'CONFIRMATION_OF_PARTICIPATION')
+        if($this->getValue('cat_name_intern') == 'EVENTS')
```
Comment (ximex) [inline adm_program/system/classes/tableroles.php]: `$this->getValue('cat_name_intern') === 'EVENTS'`

File: adm_program/system/classes/componentupdate.php [hunk 5]
```diff
@@ -343,6 +343,56 @@ public function updateStepDeleteDateRoles()
...
+
+                // now set name intern explicit to EVENTS
+                $category->setValue('cat_name_intern', 'EVENTS');
+                $category->save();
+
+                // all existing events of this organization must get the new category
+                $sql = 'UPDATE '.TBL_ROLES.' SET rol_cat_id = '.$category->getValue('cat_id').'
+                         WHERE rol_id IN (SELECT dat_rol_id
+                                            FROM '.TBL_DATES.'
+                                           INNER JOIN '.TBL_CATEGORIES.' ON cat_id = dat_cat_id
```
Comment (ximex) [inline adm_program/system/classes/componentupdate.php]: put the `ON ...` in a new line and align the `M` from `FORM` with the `N` from `INNER JOIN`

File: adm_program/modules/lists/lists.php [hunk 6]
```diff
@@ -32,6 +39,17 @@
     $headline = $gL10n->get('LST_INACTIVE_ROLES');
 }
 
+// only users with the right to assign roles can view inactive roles
+// within PHP 5.3 false will not be set and therefore we must add 0 as value
+if($getActiveRole || !$gCurrentUser->checkRolesRight('rol_assign_roles'))
+{
+    $getActiveRole  = 1;
```
Comment (ximex) [inline adm_program/modules/lists/lists.php]: `$getActiveRole = 1;`

File: adm_program/modules/lists/lists.php [hunk 7]
```diff
@@ -32,6 +39,17 @@
...
+}
+else
+{
+    $getActiveRole  = 0;
```
Comment (ximex) [inline adm_program/modules/lists/lists.php]: `$getActiveRole = 0;`

File: adm_program/modules/dates/dates_function.php [hunk 8]
```diff
@@ -387,7 +387,7 @@
             // Kategorie fuer Terminbestaetigungen einlesen
             $sql = 'SELECT cat_id
                       FROM '.TBL_CATEGORIES.'
-                     WHERE cat_name_intern = \'CONFIRMATION_OF_PARTICIPATION\'';
+                     WHERE cat_name_intern LIKE \'EVENTS\'';
```
Comment (ximex) [inline adm_program/modules/dates/dates_function.php]: only use `LIKE` if you search for `%XXX%` or something else
`WHERE cat_name_intern = \'EVENTS\'';`

File: adm_program/modules/lists/mylist.php [hunk 9]
```diff
@@ -27,8 +27,9 @@
 $getActiveRole  = admFuncVariableIsValid($_GET, 'active_role',  'bool', array('defaultValue' => true));
 $getShowMembers = admFuncVariableIsValid($_GET, 'show_members', 'int');
 
+// only users with the right to assign roles can view inactive roles
 // within PHP 5.3 false will not be set and therefore we must add 0 as value
-if($getActiveRole)
+if($getActiveRole || !$gCurrentUser->checkRolesRight('rol_assign_roles'))
 {
     $getActiveRole  = 1;
```
Comment (ximex) [inline adm_program/modules/lists/mylist.php]: `$getActiveRole = 1;`

File: adm_program/modules/dates/dates_function.php [hunk 10]
```diff
@@ -386,7 +386,7 @@
             // Kategorie fuer Terminbestaetigungen einlesen
             $sql = 'SELECT cat_id
                       FROM '.TBL_CATEGORIES.'
-                     WHERE cat_name_intern LIKE \'CONFIRMATION_OF_PARTICIPATION\'';
+                     WHERE cat_name_intern LIKE \'EVENTS\'';
```
Comment (ximex) [inline adm_program/modules/dates/dates_function.php]: `WHERE cat_name_intern = \'EVENTS\'';`

File: adm_program/modules/roles/roles.php [hunk 11]
```diff
@@ -58,20 +58,26 @@
...
-    $visibleRolesFlag  = '0';
-    // in invisible mode show active and inactive invisible roles
+    if($gPreferences['enable_dates_module'] == 0)
+    {
+        $gMessage->show($gL10n->get('SYS_NO_RIGHTS'));
+        // => EXIT
+    }
+
+    $eventsRolesLinkDescription = $gL10n->get('ROL_ACTIV_ROLES');
+    $listDescription   = $gL10n->get('ROL_ROLES_CONFIRMATION_OF_PARTICIPATION');
```
Comment (ximex) [inline adm_program/modules/roles/roles.php]: align the `=` with the other lines

File: adm_program/system/classes/user.php [hunk 12]
```diff
@@ -1619,6 +1619,29 @@ public function updateLoginData()
...
     /**
+     * Checks the necessary rights if this user could view former roles members. Therefore
+     * the user must also have the right to view the role. So you must also check this right.
+     * @return bool Return @b true if the user has the right to view former roles members
+     */
+    public function viewFormerRolesMembers()
+    {
+        global $gPreferences;
+
+        $returnCode = false;
```
Comment (ximex) [inline adm_program/system/classes/user.php]: remove it. see comments below

**Comment by ximex** [comment]
more review will come in the next days

**Comment by Fasse** [comment]
merge?

**Pull request merged**